applications/portal/frontend/src/components/header/Header.tsx

```diff
@@ -107,16 +108,24 @@ export const Header = ({
                            user,
                        }: HeaderProps) => {
     const classes = useStyles();
-
+    const api = workspaceService.getApi();
     const match = useRouteMatch<RouteParams>(EXPERIMENTS_ROUTE);
     const experimentId = match ? match.params.id : null;
 
     const [modalKey, setModalKey] = React.useState(0)
     const [dialogOpen, setDialogOpen] = React.useState(false);
     const [experimentDialogOpen, setExperimentDialogOpen] = React.useState(false);
     const [errorMessage, setErrorMessage] = React.useState(undefined);
+    const [experiments, setExperiments] =  React.useState([]);
     const menuAnchorRef = React.useRef(null);
 
+    const fetchExperiments = async () => {
+        const response = await api.listExperiments()
+        setExperiments(response.data)
+    }
+
+    const experiment = experiments.find((experiment) => experiment.id == experimentId);
```

Comment:
As a rule of thumb, when possible, it's better to have the backend to do the filtering instead of the frontend.

You can achieve the same as the two instructions above by calling 
`api.retrieveExperiment(experimentId)`